### ISSUE TITLE
setErrorHandler - not returning custom error.

### ISSUE THREAD

**Issue by Furtini** [fastify/help#197]
#### What are you trying to achieve or the steps to reproduce?

Hello, 
I'm trying to use the setErrorHandler API to return a custom error object back to the clients. 
So I throw a custom error object in my controller, the setErrorHandler catches the error, but when I send it using the reply.send(CustomError), the error that the client receives is the default error build by fastify.  

I was expecting to get the custom error object.
Even if I don't use the setErrorHandler, I'm still not getting the custom error object.

I`m also not using fastify-sensible.

I'm new here and using fastify, so I'm probably doing something wrong, but I ran out of ideas.
Any help?
Thanks

*controller.js*
```js
 async create (request) {
    const { body } = request
     try {
      const objID = await this.service.create(body)
      return { objID }
    } catch (err) {
      throw new CustomError(err)
    }
}
```

*CustomError.js*
```js
class CustomError extends Error {
  constructor (message) {
    super(message || DEFAULT_MESSAGE)
    this.code = 400
    this.name = 'resource_already_exists'
    this.source = 'custom error'
  }
}
```

index.js
```js
const app = fastify(config.options)

app.setErrorHandler(function (error, request, reply) {
    console.warn(error) // Here it logs the custom error object correctly.
    reply.send(error)
})
```

#### What was the result you received?
```js
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "duplicate key error"
}
```
#### What did you expect?
```js
{
  "name": "resource_already_exists",
  "source": "custom error",
  "code": 400
  "message": "duplicate key error",
}
```

#### Context
* *node version*: 12.13.0
* *fastify version*: >=2.11.0
* *os*: Mac

**Comment by Eomm** [comment]
Why do you expect so? - just to understand if docs have some misunderstanding.

The `send()` recognize the Error and parse it by default as this:

https://github.com/fastify/fastify/blob/2.x/lib/reply.js#L472

To send what you want, you need to write a serializer - this is just an example to archive your needs:

```
  reply.code(400).send({ ...error })
```

---- 

Maybe it could be good feature to add:

if the Error object has a `toJSON()` function we could call it instead of parse as the default

**Comment by Furtini** [comment]
Thank you very much. 
That worked for me.
I did not think about serializing it.

I got confused because in the Reply.send() documentation says that if I pass an error object to it, Fastify will automatically create and error structure as the example shows. 

- [https://www.fastify.io/docs/latest/Reply/#errors](url)

But if I want to fully customize my error handler, I should use the setErrorHandler API.
So I thought that I could return the custom error object there, but it also uses the Reply.send() to send back the error. 
So I got confused about how to return the custom error object.

Thank you again!
